django_stubs_ext/django_stubs_ext/aliases.py

```diff
@@ -2,9 +2,15 @@
 
 if typing.TYPE_CHECKING:
     from django.db.models.query import _T, _QuerySet, _Row
+    from django.utils.functional import StrOrPromise, StrPromise
 
     ValuesQuerySet = _QuerySet[_T, _Row]
+    StrOrPromise = StrOrPromise
+    StrPromise = StrPromise
```

Comment:
This isn’t [how re-exports work](https://mypy.readthedocs.io/en/stable/config_file.html#confval-implicit_reexport) in mypy.

But I think your initial inclination to use leading underscores in `django-stubs/utils/functional.pyi` was correct, because these don’t exist in `django.utils.functional` at runtime. We should use underscores in `django-stubs` and re-export with no underscores here in `django_stubs_ext`.